server/applicationInfo.ts

```diff
@@ -2,11 +2,30 @@ import fs from 'fs'
 import path from 'path'
 import config from './config'
 
-const packageJson =
-  process.env.NODE_ENV !== 'test' ? path.join(__dirname, '../../package.json') : path.join(__dirname, '../package.json')
+const { buildNumber, gitRef, productId, branchName } = config
 
-const { name: applicationName } = JSON.parse(fs.readFileSync(packageJson).toString())
+export type ApplicationInfo = {
+  applicationName: string
+  buildNumber: string
+  gitRef: string
+  gitShortHash: string
+  productId?: string
+  branchName: string
+}
 
-const { buildNumber, gitRef } = config
+const getNameFromPackageJson = () => {
+  const packageJson = path.join(__dirname, '../../package.json')
+  const { name } = JSON.parse(fs.readFileSync(packageJson).toString())
+  return name
+}
 
-export default { applicationName, buildNumber, gitRef, gitShortHash: gitRef.substring(0, 7) }
+export default (overrideName?: string): ApplicationInfo => {
+  return {
+    applicationName: overrideName || getNameFromPackageJson(),
```

Comment:
little optimisation to only load name from the file system if no override was specified